lib/practice_exercises.rb

```diff
@@ -1,13 +1,34 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# I have a loop that's dependant on the length of the string.
+# Space Complexity: O(n) because I'm creating a new string based on the input.
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string = string.gsub(/[^0-9a-zA-Z]/, '').downcase
+  first = 0
+  last = string.length - 1
+  until first >= last
+    return false if string[first] != string[last]
+    first += 1
+    last -= 1
+  end
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n * m) where n is the length of an array and m is the length of a given string
+# worst case would be O(n^2), however if n becomes sufficiently large compared to m
+# O(n) would be more realistic.
+# Space Complexity: O(1), not creating new structures, just using variables & integers.
 def longest_prefix(strings)
```

Comment:
You are returning a substring of the shortest string in the list, so this has space complexity O(m).  You also have two nested loops depending on the number of strings and the length of the 1st string so this is O(nm) time complexity.